Dockerfile

```diff
@@ -17,11 +17,21 @@ MAINTAINER devops@edx.org
 
 # libmysqlclient-dev; to install header files needed to use native C implementation for 
 # MySQL-python for performance gains.
-
+# software-properties-common; to get apt-add-repository
 # If you add a package here please include a comment above describing what it is used for
-RUN apt-get update && apt-get upgrade -qy && apt-get install language-pack-en locales git python3.5 python3-pip libmysqlclient-dev libssl-dev python3-dev -qy && \
-pip3 install --upgrade pip setuptools && \
-rm -rf /var/lib/apt/lists/*
+
+RUN apt-get update &&\
+    apt-get install -y locales &&\
+    apt-get install -y locales software-properties-common &&\
+    locale-gen en_US.UTF-8
+
+# Add the deadsnakes PPA to install Python 3.8
+RUN apt-add-repository -y ppa:deadsnakes/ppa
+
+RUN apt-get update && apt-get upgrade -qy && apt-get install language-pack-en locales git python3.6 python3-pip libmysqlclient-dev libssl-dev python3-dev -qy && \
```

Comment:
I think we may want to leave all apt-get installs on one RUN command to keep the image size smaller and split each package into a new line to make future diffs more readable?

Reply:
We need software-properties-common to get apt-add-repository that we can use to add deadsnakes ppa. This is I need a separate RUN for installation.

Reply:
That makes sense.  I think we should eventually run build separate images for each version of Python, but for now this seems ok.  Could we include `&&  rm -rf /var/lib/apt/lists/*` at the end of the `RUN apt-get` instructions to clean up the apt cache so it isn’t stored in the layer. See more in the Docker Docs under APT here: https://docs.docker.com/develop/develop-images/dockerfile_best-practices

Another option would be to squash all of the apt layers into one by &&'ing line 23-34 into one run line so we don't store the intermediary layers.

Reply:
@adzuci Combining multiple RUN statements image size is now ```820MB```